File: files/2025-10-12/CSEC 12102025.py
import pandas as pd

path = "files/2025-10-12/Challenge 70.xlsx"
input = pd.read_excel(path, usecols="B:C", skiprows=1, nrows=11)
test = pd.read_excel(path, usecols="E:F", skiprows=1, nrows=4).rename(columns=lambda col: col.replace('.1', ''))

input['Item'] = "Group " + input['Item'].str.contains("Group").cumsum().astype(str)
result = input.groupby('Item', as_index=False)['Prices'].sum()

print(result.equals(test))

File: files/2025-10-12/CSEC 12102025.R
library(tidyverse)
library(readxl)

path = "files/2025-10-12/Challenge 70.xlsx"
input = read_excel(path, range = "B2:C12")
test  = read_excel(path, range = "E2:F6")

result = input %>%
  mutate(Item = paste0("Group ",cumsum(str_detect(Item, "Group")))) %>%
  summarise(Prices = sum(Prices,na.rm = TRUE), .by = Item)

all.equal(result, test)
# [1] TRUE